atlp-rwanda/e-comm-team-techsmith-fn
# #185218929 Users should be able to chat

## What does this PR do?

Adds chat and chat room features to our ECommerce application.

## Description of Task to be completed?

+ Given a user (buyer) is logged in and online, they should be able to share a public
+ Given a user opens the chat tab, all fellow active users are notified of a new party joining the chat
+ When a user sends a message, it is echoed on all active participants
+ All messages are stored in the database
+ A user should be able to read all past messages from the chat

## How should this be manually tested?

- Clone the repository using `git clone https://github.com/atlp-rwanda/e-comm-team-techsmith-fn`

- Checkout to the branch using `git checkout ft-chatroom-185218929`

- Install node packages that the application depends on: `npm i`

- Create a `.env` file in the root directory and add all environment variables required to run the application. The required variables are listed in the `.env.example` file available in root directory of the application.

- Start the application development server: `npm start`

- Check out the components documentation: `npm run storybook`

## What are the relevant pivotal tracker/Trello stories?

+ [#185218929](https://www.pivotaltracker.com/story/show/185218929)

## Screenshots (if applicable)
<img width="1440" alt="Screenshot 2023-06-16 at 15 37 26" src="https://github.com/atlp-rwanda/e-comm-team-techsmith-fn/assets/31816352/aafbf5d9-50c7-499a-9eb3-8fb098a453c4">
<img width="1436" alt="Screenshot 2023-06-17 at 17 25 09" src="https://github.com/atlp-rwanda/e-comm-team-techsmith-fn/assets/31816352/56fb08b6-f349-41ef-9557-ee551d594ff0">
<img width="1436" alt="Screenshot 2023-06-17 at 17 25 47" src="https://github.com/atlp-rwanda/e-comm-team-techsmith-fn/assets/31816352/bc023931-81d4-45f5-8cb9-80de4b1e9bfe">
<img width="1436" alt="Screenshot 2023-06-17 at 17 25 58" src="https://github.com/atlp-rwanda/e-comm-team-techsmith-fn/assets/31816352/f0a40a34-3936-49e9-9042-79d98e3af4f8">
<img width="1430" alt="Screenshot 2023-06-17 at 17 27 34" src="https://github.com/atlp-rwanda/e-comm-team-techsmith-fn/assets/31816352/968fd476-b639-4044-a7e0-406f6a698ba5">
